htdocs/mrp/mo_card.php

```diff
@@ -176,6 +176,66 @@
 		exit;
 	}
 
+	if ($action == 'confirm_delete' && !empty($permissiontodelete)) {
+		if (!($object->id > 0)) {
+			dol_print_error('', 'Error, object must be fetched before being deleted');
+			exit;
+		}
+
+		$error = 0;
+		$deleteChilds = GETPOST('deletechilds', 'boolean');
+
+		if ($deleteChilds === 'on') {
+			$TMoChildren = $object->getAllMoChilds();
+
+			// Supprimer les objets enfants en utilisant une transaction séparée
+			$db->begin();
```

Comment:
The delete of the child and the delete of the parent (line 212) must be included into the same transaction.